### PR TITLE
Update IStreamHandler and label_sample

### PR DESCRIPTION
**Pull request by assapir**

### CODE REVIEW

File: sdk/file_sdk/api_impl/stream_handler.cpp [hunk 1]
```diff
@@ -68,14 +66,22 @@ void StreamHandler::SetProtection(const UserPolicy& policy) {
 void StreamHandler::RemoveProtection() {
 }
 
-void StreamHandler::Commit(std::shared_ptr<IStream> outputStream, std::string& outputExtension) {
-  mFileFormat->UpdateProperties(mProperties, {});
-  mFileFormat->Commit(outputStream, outputExtension);
-}
-
 time_t StreamHandler::GetCurrentTime() {
```
Comment (shenhavba) [inline sdk/file_sdk/api_impl/stream_handler.cpp]: please change it to use the correct time format like i did, the sample should show the correct way to use the sdk...
Also, it should be UTC time #Resolved

File: sdk/file_sdk/api/istream_handler.h [hunk 2]
```diff
@@ -28,18 +28,16 @@ class IStreamHandler {
 
   virtual std::shared_ptr<Tag> GetLabel() = 0;
 
-  virtual void SetLabel(const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
```
Comment (shenhavba) [inline sdk/file_sdk/api/istream_handler.h:31]: i think it should be an out parameter #Resolved

File: sdk/file_sdk/api/istream_handler.h [hunk 3]
```diff
@@ -28,18 +28,16 @@ class IStreamHandler {
...
+  virtual std::string SetLabel(std::shared_ptr<IStream> outputStream, const std::string& labelId, const LabelingOptions& labelingOptions) = 0;
 
-  virtual void SetLabel(std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
+  virtual std::string SetLabel(std::shared_ptr<IStream> outputStream, std::shared_ptr<ILabel> label, const LabelingOptions& labelingOptions) = 0;
 
-  virtual void DeleteLabel(const std::string& justificationMessage) = 0;
+  virtual std::string DeleteLabel(std::shared_ptr<IStream> outputStream, const std::string& justificationMessage) = 0;
```
Comment (vakuras) [inline sdk/file_sdk/api/istream_handler.h]: change to out parameter #Resolved

File: sdk/file_sdk/api/istream_handler.h [hunk 4]
```diff
@@ -28,18 +28,16 @@ class IStreamHandler {
...
 
   virtual void SetProtection(const UserPolicy& policy ) = 0;
```
Comment (vakuras) [inline sdk/file_sdk/api/istream_handler.h]: change to the new class #Resolved

Comment (assapir) [reply sdk/file_sdk/api/istream_handler.h]: next PR to seprate to P and LP handler

---
In reply to: [141048268](https://github.com/AzureAD/rms-sdk-for-cpp/pull/170#discussion_r141048268) [](ancestors = 141048268)

Comment (vakuras) [reply sdk/file_sdk/api/istream_handler.h]: Let's do the move to MIP first before you start seperating it. #Resolved

**Review by vakuras — APPROVED** [review]
I'll do a more through review after you move the code to MIP repo and connect it with the PolicyEngine.

**Pull request merged**